pkg/gardenadm/botanist/kubelet_test.go

```diff
@@ -107,4 +110,23 @@ var _ = Describe("Kubelet", func() {
 			Entry("with existing token file", false),
 		)
 	})
+
+	Describe("#BootstrapKubelet", func() {
+		It("should do nothing when the node was already found", func() {
+			Expect(fakeSeedClient.Create(ctx, &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "foo", Labels: map[string]string{"kubernetes.io/hostname": hostName}}})).To(Succeed())
+
+			Expect(b.BootstrapKubelet(ctx)).To(Succeed())
+		})
```

Comment:
Should we add some check to ensure that actually nothing was done, e.g. checking for one of the cluster role bindings or the bootstrap token?